Nginx template improvements

MarekBenjamin opened the pull request:
- Updated sample-nginx.conf to use php7.4-fpm (was 7.0)
- Removed double hard coded friendica.example.net declatation by replacing defining it with already defined `$server_name` variable
- Added `sample-nginx-cerbot.config` to be used directly with `cerbot --nginx`

Commit:
Replaced hostname in rewrite url for forwarding http to https by $server_name variable instead of hard coded example.com url

Updated default PHP-FPM path from 7.0 to 7.4

Commit:
Added sample-nginx-cerbot.config to easly configure nginx to use ssl for friendica using certbot

MrPetovan reviewed (commented):
Is the `mods/sample-nginx-certbot.config` file intended to be empty?

MarekBenjamin commented:
Not at all! Gonna check that shortly

24.11.2022 20:45:42 Hypolite Petovan ***@***.***>:

> ***@***.**** commented on this pull request.
> 
> Is the *mods/sample-nginx-certbot.config* file intended to be empty?
> 
> —
> Reply to this email directly, view it on GitHub[https://github.com/friendica/friendica/pull/12259#pullrequestreview-1193565319], or unsubscribe[https://github.com/notifications/unsubscribe-auth/A4CPKZRMHUFQDXSTU4R4C5TWJ7AWHANCNFSM6AAAAAASKVUGXU].
> You are receiving this because you authored the thread.[Verfolgungsbild][https://github.com/notifications/beacon/A4CPKZSZGQHLQJIVNZTONCDWJ7AWHA5CNFSM6AAAAAASKVUGXWWGG33NNVSW45C7OR4XAZNRKB2WY3CSMVYXKZLTORJGK5TJMV32UY3PNVWWK3TUL5UWJTSHEROIO.gif]
>

Commit:
content of sample-nginx-certbot.config

MarekBenjamin committed:
Merge branch 'friendica:develop' into nginx_template_improvements

MarekBenjamin commented:
Updated with the actual content,

nupplaphil commented on mods/sample-nginx-certbot.config:
```diff
@@ -0,0 +1,128 @@
+##
+# Friendica Nginx configuration template to be autocnfgiured with cerbot
+# nased on sample-nginx.config by Olaf Conradi
```
```suggestion
# Friendica Nginx configuration template to be autoconfigured with certbot
# based on sample-nginx.config by Olaf Conradi
```

MarekBenjamin committed:
Update mods/sample-nginx-certbot.config

Damn typos, I am too used to use autocorrect, need to learn again to proofread my own texts! Thank you very much! <3

Co-authored-by: Philipp <admin+Github@philipp.info>

Commit:
typo

Commit:
Merge remote-tracking branch 'origin/nginx_template_improvements' into nginx_template_improvements

nupplaphil commented:
Thank you for your work :-)
